### PR TITLE
Reworked Docs landing page

### PR DESCRIPTION
**Pull request by chalin**
Contributes to #256

(More changes are coming, but this seems like enough of an increment to publish.)

cc @thisisnotapril

### CODE REVIEW

File: content/docs/_index.md [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
+
+- **New to gRPC?** Start with the following pages:
+
+  - [What is gRPC?](guides)
+  - [gRPC Concepts](guides/concepts)
+  - [FAQ](/faq)
+
+- **Eager to see gRPC in action?**
+
+  [Select a language](languages) and then choose it's **Quick Start**.
```
Comment (lucperkins) [inline content/docs/_index.md:21]: its

File: content/docs/_index.md [hunk 2]
```diff
@@ -1,21 +1,28 @@
...
-work underway to develop a JS library for use in browsers. Beyond that, it is
-ideal as a microservices interconnect, not just because the core protocol is
-very efficient but also because the framework has pluggable authentication, load
-balancing, etc. Google itself is also transitioning to use it to connect
-microservices.
+Learn about key gRPC concepts, try a quick start, find tutorials and reference
+material for all [supported languages](languages).
+
+
+<style>
```
Comment (lucperkins) [inline content/docs/_index.md:9]: Inline CSS should be avoided

Comment (chalin) [reply content/docs/_index.md:9]: I agree, though this page will likely be changing again soon in a way that will make this style tweak unnecessary.

Comment (chalin) [reply content/docs/_index.md:9]: I'm not that familiar with Bulma, maybe you can suggest an alternative way to achieving this? Thanks

Comment (lucperkins) [reply content/docs/_index.md:9]: If you intend to remove this, it's fine.

File: content/docs/_index.md [hunk 3]
```diff
@@ -1,21 +1,28 @@
...
+
+- **Eager to see gRPC in action?**
+
+  [Select a language](languages) and then choose its **Quick Start**.
+
+- Interested in **gRPC feature details?**
+
+  Try one of the following:
+
+  - [Select a language](languages) and then choose a **tutorial** or **API reference**
```
Comment (mjpitz) [inline content/docs/_index.md:27]: The `choose a ...` makes this a little awkward to read. Here are some alternatives

* Select a language and then choose either tutorial or API reference
* Select a language and then choose from tutorial or API reference
* Select a language and then choose tutorial or API reference

File: content/docs/_index.md [hunk 4]
```diff
@@ -1,21 +1,28 @@
 ---
-title: Overview
+title: gRPC Documentation
```
Comment (mjpitz) [inline content/docs/_index.md:2]: Should this just be "Documentation"? gRPC feels a little repetative.

Comment (chalin) [reply content/docs/_index.md]: Done

File: content/docs/_index.md (hunk 3 above)
Comment (chalin) [reply content/docs/_index.md:26]: Going with your last suggestion.

File: content/docs/_index.md [hunk 5]
```diff
@@ -1,21 +1,27 @@
...
+Learn about key gRPC concepts, try a quick start, find tutorials and reference
+material for all [supported languages](languages).
+
+<style>
+  div > ul > li { padding-top: 1em !important; }
+</style>
+
+- **New to gRPC?** Start with the following pages:
+
+  - [What is gRPC?](guides)
```
Comment (ejona86) [inline content/docs/_index.md:14]: This link looks wrong.

File: content/docs/_index.md [hunk 6]
```diff
@@ -1,21 +1,27 @@
...
+  - [gRPC Concepts](guides/concepts)
```
Comment (ejona86) [inline content/docs/_index.md:15]: And this looks to be the old link (so it redirects).

Comment (chalin) [reply content/docs/_index.md:15]: Hmm, I was sure that I had fixed those. Thanks for the careful read.

File: content/docs/_index.md (hunk 2 above)
Comment (chalin) [reply content/docs/_index.md:8]: Inlined styling removed via #343.

**Pull request merged**